caom2repo/caom2repo/tests/test_core.py

```diff
@@ -516,8 +575,124 @@ def test_process(self):
                               'MAXREC': 3})]
         visitor._repo_client.get.assert_has_calls(calls)
 
+    def mock_get_observation(self, collection, observationID):
+        return SimpleObservation(collection, observationID)
+
+    def test_multiprocess_with_obs_id(self):
+        with patch('caom2repo.core.CAOM2RepoClient', PickableMagicMock()) as client_mock:
```

Comment:
I would probably use the decorator format since the mock applies to the entire function